gpkit/constraints/relax.py

```diff
@@ -0,0 +1,142 @@
+"""Models for assessing primal feasibility"""
+from .set import ConstraintSet
+from ..nomials import Variable, VectorVariable, parse_subs, NomialArray
+from ..nomials import Monomial
+from ..keydict import KeyDict
+from .. import NamedVariables, MODELNUM_LOOKUP
+
+
+class ConstraintsRelaxedEqually(ConstraintSet):
+    """Relax constraints the same amount, as in Eqn. 10 of [Boyd2007].
+
+    Arguments
+    ---------
+    constraints : iterable
+        Constraints which will be relaxed (made easier).
+
+    varname : str
+        LaTeX name of relaxation variable.
+
+
+    Attributes
+    ----------
+    relaxvar : Variable
+        The variable controlling the relaxation. A solved value of 1 means no
+        relaxation. Higher values indicate the amount by which all constraints
+        have been made easier: e.g., a value of 1.5 means all constraints were
+        50 percent easier in the final solution than in the original problem.
+
+    [Boyd2007] : "A tutorial on geometric programming", Optim Eng 8:67-122
+
+    """
+
+    def __init__(self, constraints):
+        if not isinstance(constraints, ConstraintSet):
+            constraints = ConstraintSet(constraints)
+        substitutions = dict(constraints.substitutions)
+        posynomials = constraints.as_posyslt1()
+        with NamedVariables("Relax"):
+            self.relaxvar = Variable("C")
+        ConstraintSet.__init__(self,
+                               [[posy <= self.relaxvar
+                                 for posy in posynomials],
+                                self.relaxvar >= 1],
```

Comment:
I still question whether the `self.relaxvar >= 1` constraint is necessary (omitting it would allow feasible models to still be pressured in the direction of "make all constraints as feasible as possible"; the relax var is pressured down by the objective and up by all constraints)